File: frontend/src/services/quizService.js
import apiClient from './api';

export default {
  // Start a new quiz session
  startQuiz() {
    return apiClient.post('/quiz/start');
  },

  // Start a review session with wrong questions
  startReview() {
    return apiClient.post('/quiz/review');
  },

  // Get the current question
  getCurrentQuestion(index = 0, mode = 'normal') {
    return apiClient.get('/quiz/current', { params: { index, mode } });
  },

  // Submit an answer
  submitAnswer(questionId, userAnswer, mode = 'normal') {
    return apiClient.post('/quiz/submit', { questionId, userAnswer, mode });
  },

  // Get the next question
  getNextQuestion(currentIndex, mode = 'normal') {
    return apiClient.get('/quiz/next', { params: { currentIndex, mode } });
  },

  // Get the previous question
  getPreviousQuestion(currentIndex, mode = 'normal') {
    return apiClient.get('/quiz/previous', { params: { currentIndex, mode } });
  },

  // Get user answer history
  getUserAnswers(filters = {}, page = 1, limit = 100) {
    return apiClient.get('/quiz/answers', { 
      params: { page, limit, ...filters } 
    });
  },

  // Get answers for a specific question
  getQuestionAnswers(questionId) {
    return apiClient.get(`/quiz/answers/${questionId}`);
  },

  // Reset quiz progress
  resetQuiz() {
    return apiClient.post('/quiz/reset');
  },

  // Get navigation data for questions
  getQuestionNavigator(filter = 'all', type = 'all', search = '') {
    return apiClient.get('/quiz/navigator', {
      params: { filter, type, search }
    });
  },

  // Jump to a specific question
  jumpToQuestion(index, mode = 'normal') {
    return apiClient.post(`/quiz/jump/${index}`, { mode });
  }
};

File: backend-springboot/src/main/java/com/quizgen/app/quiz/service/QuizService.java
package com.quizgen.app.quiz.service;

import com.quizgen.app.quiz.dto.StartQuizRequest;
import com.quizgen.app.quiz.dto.SubmitAnswerRequest;

import java.util.List;
import java.util.Map;

public interface QuizService {

    Map<String, Object> startQuiz(StartQuizRequest request);

    Map<String, Object> startReview();

    Map<String, Object> startByQuestionIds(String mode, List<Long> questionIds);

    Map<String, Object> getCurrentQuestion(int index, String mode);

    Map<String, Object> nextQuestion(int currentIndex, String mode);

    Map<String, Object> previousQuestion(int currentIndex, String mode);

    Map<String, Object> submitAnswer(SubmitAnswerRequest request);

    Map<String, Object> getUserAnswers(int page, int limit, String isCorrect, String sessionType);

    List<Map<String, Object>> getQuestionAnswers(String questionCode);

    void resetQuiz();

    List<Map<String, Object>> navigator(String filter, String type, String search);

    Map<String, Object> jumpToQuestion(int index, String mode);
}

